src/sources/socket/udp.rs

```diff
@@ -55,7 +65,12 @@ pub fn udp(address: SocketAddr, host_key: Atom, out: mpsc::Sender<Event>) -> Sou
                 .map_err(|error: io::Error| error!(message = "error reading datagram.", %error))
                 .forward(out)
                 // Done with listening and sending
-                .map(|_| ())
+                .map(|_| {
```

Comment:
For example, we could find a way to store this "oneshot" handle along side this returned future and when it exists we can then notify the main tripwire.